python/ray/ml/integrations/torch/torch_utils.py

```diff
@@ -0,0 +1,109 @@
+from typing import Optional, Union, Dict, List
+
+import pandas as pd
+import torch
+
+
+def convert_pandas_to_torch_tensor(
+    data_batch: pd.DataFrame,
+    columns: Optional[Union[List[str], List[List[str]]]] = None,
+    column_dtypes: Optional[Union[torch.dtype, List[torch.dtype]]] = None,
+) -> Union[torch.Tensor, List[torch.Tensor]]:
+
+    """Converts a Pandas dataframe to a torch Tensor or list of torch Tensors.
+
+    The format of the return type will match the format of ``columns``. If a
+    list of columns is provided, the return type will be a single tensor. If
+    ``columns`` is a list of list, then the return type will be a list of
+    tensors.
+
+    Args:
+        data_batch (pandas.DataFrame): The pandas dataframe to conver to a
+            torch tensor.
+        columns (Optional[Union[List[str], List[List[str]]]):
+            The names of the columns in the dataframe to include in the
+            torch tensor. If this arg is a List[List[str]], then the return
+            type will be a List of tensors. This is useful for multi-input
+            models. If None, then use all columns in the ``data_batch``.
+        column_dtype (Optional[Union[torch.dtype, List[torch.dtype]): The
+            torch dtype to use for the tensor. If set to None,
+            then automatically
+            infer the dtype.
+
+    Returns:
+        Either a torch tensor of size (N, len(columns)) where N is the
+        number of rows in the ``data_batch`` Dataframe, or a list of
+        tensors, where the size of item i is (N, len(columns[i])).
+
+    """
```

Comment:
Passing a list of dtypes doesn't work, as the resulting tensor will have one dtype only (the biggest one). Therefore, only one dtype should be passed.